### PR TITLE
Update core modules to fix strict mode errors

### PR DESCRIPTION
**Pull request by cchurch**
- Also fix creates parameter issue in win_msi as described in https://github.com/ansible/ansible-modules-core/issues/129, slightly different fix from https://github.com/ansible/ansible-modules-core/pull/1482
- Fix setup.ps1 module issue described in https://github.com/ansible/ansible-modules-core/issues/1927

Needed to support StrictMode changes in https://github.com/ansible/ansible/pull/12047

### CODE REVIEW

**Comment by gregdek** [comment]
Looks like a pretty big patch... reviews welcome from the Windows crew.

Ping @trondhindenes @petemounce @elventear @smadam813 @jhawkesworth @angstwad @sivel @chrishoffman

File: windows/win_msi.ps1 [hunk 1]
```diff
@@ -21,36 +21,28 @@
...
-    }
+    Exit-Json $result;
 }
 
 $logfile = [IO.Path]::GetTempFileName();
-if ($params.state.GetType -and $params.state -eq "absent")
+if ($state -eq "absent")
 {
-    msiexec.exe /x $params.path /qb /l $logfile $extra_args;
+    msiexec.exe /x $path /qn /l $logfile $extra_args
```
Comment (petemounce) [inline windows/win_msi.ps1:41]: Why `/qb` -> `/qn`?

**Comment by petemounce** [comment]
otherwise, lgtm - shipit.

File: windows/win_msi.ps1 (hunk 1 above)
Comment (cchurch) [inline windows/win_msi.ps1:41]: Just something I noticed when looking through the module.  `/qb` displays a basic UI, `/qn` displays no UI, which seems more appropriate in this case.

**Comment by trondhindenes** [comment]
Nice work @cchurch. +1 for merge.

**Comment by msghaleb** [comment]
Nice work @cchurch I've one comment. If I try to use the module now, its just giving me an OK even if it installs the MSI file.

It seems not to set the $result.changed to $true in line 48, changing it from "set-Attr $result "changed" $true" to be just $result.changed = $true fixed it (see below)

48 #Set-Attr $result "changed" $true;
49 $result.changed = $true;

Regards,

MG

**Comment by cchurch** [comment]
@msghaleb Can you confirm you also have the changes from ansible/ansible#12384 in your checkout?  There was a fix added in that one for `Set-Attr`; previously it wouldn't actually set the attribute if it had already been set.

**Comment by msghaleb** [comment]
@cchurch No I don't have #12384 changes in my checkout. I've imported only win_msi.ps1/py to my repo.
I'll include the changes and update you back if it didn't work.

**Pull request merged**